Sources/NIO/ConvenienceOptionSupport.swift

```diff
@@ -0,0 +1,440 @@
+            toReturn = toReturn.channelOption(option)
+        }
+        return toReturn
+    }
+    
+    @usableFromInline
+    func channelOption(_ option: NIOPipeShorthandOption) -> NIOPipeBootstrap {
+        let applier = NIOPipeBootstrap_Applier(contained: self)
+        return option.applyOptionDefaultMapping(with: applier).contained
+    }
+    
+    fileprivate struct NIOPipeBootstrap_Applier : NIOChannelOptionAppliable {
+        var contained : NIOPipeBootstrap
+        
+        func applyOption<Option: ChannelOption>(_ option: Option, value: Option.Value) -> NIOPipeBootstrap_Applier {
+            return NIOPipeBootstrap_Applier(contained: contained.channelOption(option, value: value))
+        }
+    }
+}
+
+// MARK:  Universal Client Bootstrap
+extension NIOClientTCPBootstrapProtocol {
+    /// Apply a shorthand option to this bootstrap - default implementation which always fails to apply.
+    /// - parameters:
+    ///     - option:  The option to try applying.
+    /// - returns: The updated bootstrap if option was successfully applied, otherwise nil suggesting the caller try another method.
+    public func applyChannelOption(_ option: NIOTCPShorthandOption) -> Self? {
+        return .none
+    }
+}
+
+extension NIOClientTCPBootstrap {
+    /// Specifies some `ChannelOption`s to be applied to the channel.
+    /// - See: channelOption
+    /// - Parameter options: List of shorthand options to apply.
+    /// - Returns: The updated bootstrap (`self` being mutated)
+    @inlinable
+    public func channelOptions(_ options: [NIOTCPShorthandOption]) -> NIOClientTCPBootstrap {
+        var toReturn = self
+        for option in options {
+            toReturn = toReturn.channelOption(option)
+        }
+        return toReturn
+    }
+    
+    @usableFromInline
+    func channelOption(_ option: NIOTCPShorthandOption) -> NIOClientTCPBootstrap {
+        if let updatedUnderlying = underlyingBootstrap.applyChannelOption(option) {
+            return NIOClientTCPBootstrap(self, withUpdated: updatedUnderlying)
+        } else {
+            let applier = NIOClientTCPBootstrap_Applier(contained: self)
+            return option.applyOptionDefaultMapping(with: applier).contained
+        }
+    }
+    
+    struct NIOClientTCPBootstrap_Applier : NIOChannelOptionAppliable {
+        var contained : NIOClientTCPBootstrap
+        
+        func applyOption<Option: ChannelOption>(_ option: Option, value: Option.Value) -> NIOClientTCPBootstrap_Applier {
+            return NIOClientTCPBootstrap_Applier(contained: contained.channelOption(option, value: value))
+        }
+    }
+}
+
+// MARK: Utility
+/// An object which can have a 'ChannelOption' applied to it and will return an appropriately updated version of itself.
+public protocol NIOChannelOptionAppliable {
+    /// Apply a ChannelOption and return an updated version of self.
+    func applyOption<Option: ChannelOption>(_ option: Option, value: Option.Value) -> Self
```

Comment:
I'm sure we had a thread on one of the PRs. @Lukasa can't remember what we thought would be a good name, maybe `tcpChannelOption` instead of `applyOption`?

Reply:
Hmm, this is more general than that, no? We use this in the datagram bootstraps, for example.